Add purpose and use cases to README (#12)
metanorma/pubid: README.adoc

= Publication identifiers library

image:https://badge.fury.io/rb/pubid.svg["Gem Version", link="https://badge.fury.io/rb/pubid"]

Pubid is a comprehensive framework for building and working with publication identifiers. It provides a robust API for creating, modifying, parsing, exporting, and verifying document identifiers, which are crucial for various document management systems and standards compliance

This gem includes all pubid-* gems for compatibility testing and allows parsing all identifiers from one gem.

== Current use cases

> Current use cases are what we needed until now and discovered through application to pubid-* and relaton-* projects. It's not complete and will be expanding during an integration process.

Reviewer: ```suggestion
The current use cases, as listed below, are what we have needed until now and have discovered through use of pubid to pubid-*,  metanorma-*, and relaton-* projects. Work on pubid is not complete and will be expanding during the ongoing integration process.
```

> - build identifiers from scratch

Reviewer: ```suggestion
- building identifiers from scratch
```

> - modifiying identifiers

Reviewer: ```suggestion
- modifiying identifiers (e.g. basing an amendment identifier on a base identifier)
```

- parsing string identifiers
- exporting/importing to different formats or styles (hash, yaml, URN, etc.)

> - verification of compliance for identifier's representation standard (NIST PubID 1.0 for NIST, RFC 5141 for ISO URN) or using commonly discoveried patterns.

Reviewer: ```suggestion
- verification of compliance for the identifier's representation standard (NIST PubID 1.0 for NIST, RFC 5141 for ISO URN), or using commonly found patterns
```

> - verification of correctness for provided input.

Reviewer: ```suggestion
- verifying that a provided identifier is correct
```

> - comparison two provided identifiers (with excluding attributes)

Reviewer: ```suggestion
- comparing two provided identifiers (with the option to exclude specified attributes)
```

- converting legacy identifiers to updated format

== Purpose and Usage Guidelines

> The pubid library is intended to be used wherever operations on publication identifiers are required. It acts as an intermediary between users and publication identifiers, streamlining the processes of creation, modification, comparison, and attribute access. By centralizing these operations in pubid-core, we ensure that any updates to identifier standards are seamlessly integrated across all related projects.

Reviewer: ```suggestion
The pubid library is intended to be used wherever operations on publication identifiers are required. It acts as an intermediary between users and publication identifiers, streamlining the processes of creation, modification, comparison, and attribute access. By centralizing these operations in pubid-core, we ensure that any updates to identifier standards are seamlessly integrated across all related projects in an ecosystem.
```

> It is crucial to use the pubid library for all operations with identifiers to ensure that updates in identifier standards are consistently reflected in these operations. For example, if an identifier like `ISO/R 657/IV` is updated to `ISO/R 657-4:1969`, the library will automatically recognize this change. Hence, comparing:

Reviewer: ```suggestion
It is crucial to use the pubid library for all operations with identifiers to ensure that updates in identifier standards are consistently reflected in these operations, and are driven by a data representation of the identifier, rather than the identifier string. For example, if an identifier like `ISO/R 657/IV` is updated to `ISO/R 657-4:1969` by a standards body, and pubid acts on that update, the library will automatically recognize this change. Hence, comparing:
```

[source,ruby]
----
Pubid::Registry.parse("ISO/R 657/IV") == Pubid::Registry.parse("ISO/R 657-4:1969")
=> true
----
or accessing the year:
[source,ruby]
----
Pubid::Registry.parse("ISO/R 657/IV").year
=> 1969
----
will yield the updated results.

Moreover, changes in how identifiers are compared may cause some identifiers to become equal or unequal based on the latest updates. Thus, to reflect these changes accurately, the pubid library should be used for all publication identifier operations.

=== Usage

> `Pubid::Registry#parse` resolves class related to identifier (`Pubid::Iso::Identifier` for "ISO" identifiers) and returns object with parsed identifier

Reviewer: ```suggestion
`Pubid::Registry#parse` resolves the pubid class related to the identifier (`Pubid::Iso::Identifier` for "ISO" identifiers) and returns an object with the parsed identifier
```

[source,ruby]
----
require "pubid"

pubid = Pubid::Registry.parse("ISO/IEC 13213")
pubid.class
=> Pubid::Iso::Identifier::Base
pubid.publisher
=> "ISO"
pubid.copublisher
=> "IEC"
pubid.number
=> 13213
pubid.to_s
=> "ISO/IEC 13213"

> ----

Reviewer: Surely there's a lot more functionality in Pubid than this?

Author: This is just a basic example for `pubid`. More functionality in `pubid-core` and `pubid-*` gems.

Author: Added reference to `pubid-core` in usage section
